### PR TITLE
initial GH action for testgrid runs

### PR DESCRIPTION
**Pull request by paha**

### CODE REVIEW

File: .github/workflows/daily-testgrid.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    working-directory: ./testgrid/tgrun
+
+jobs:
+  testgrid-init:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: kubeconfig
```
Comment (marccampbell) [inline .github/workflows/daily-testgrid.yaml:17]: Why is this needed?  How will this work when there are > testgrid servers/

File: .github/workflows/daily-testgrid.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        go-version: 1.15.1
+
+    - name: tgrun-build
+      run: make build
+
+    - name: testgrid-cleanup
+      env:
+        KUBECONFIG: "/tmp/config"
+      run:
+        kubectl delete vmi --all
```
Comment (marccampbell) [inline .github/workflows/daily-testgrid.yaml:35]: This should be on the instance, not in the action. When we run this, we will need to have a dynamic set of more than 1 instance running these. It will be difficult to maintain this list in the action, but instead, the instance can delete them,.

File: .github/workflows/daily-testgrid.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    - name: testgrid-queue
+      env:
+        KUBECONFIG: "/tmp/config"
+        TESTGRID_API: "https://api.testgrid.staging.kurl.sh/"
+      
+      run: |
+        ./bin/tgrun queue --ref $(date "+%s") --testgrid-api $TESTGRID_API
```
Comment (marccampbell) [inline .github/workflows/daily-testgrid.yaml:43]: What if we used the last 7 of the sha here?

File: .github/workflows/deploy-testgrid.yaml [hunk 4]
```diff
@@ -16,6 +16,12 @@ jobs:
           --build-arg version=${GITHUB_SHA:0:7} \
           -t 923411875752.dkr.ecr.us-east-1.amazonaws.com/tgapi:${GITHUB_SHA:0:7} testgrid/tgapi
 
+    - run: |
+        docker build -f testgrid/tgrun/Dockerfile \
+          --build-arg version=${GITHUB_SHA:0:7} \
+          -t 923411875752.dkr.ecr.us-east-1.amazonaws.com/tgrun:latest \
+          -t 923411875752.dkr.ecr.us-east-1.amazonaws.com/tgrun:${GITHUB_SHA:0:7} testgrid/tgrun
```
Comment (marccampbell) [inline .github/workflows/deploy-testgrid.yaml:23]: testgrid should be public, can this be in Dockerhub instead of ecr?

**Review by marccampbell — CHANGES_REQUESTED** [review]

File: .github/workflows/daily-testgrid.yaml (hunk 1 above)
Comment (paha) [reply .github/workflows/daily-testgrid.yaml:17]: Good points,
I was intending to add `vmi` cleanup to the *main loop* of `tgrun` left it here until then, but yes at this point it is confusing and multiple tg clusters would be hard to support with this model.

removing this in the next commit and will get clean up managed by `tgrun`.

File: .github/workflows/daily-testgrid.yaml (hunk 2 above)
Comment (paha) [reply .github/workflows/daily-testgrid.yaml:35]: Agreed, please see the previous comment on this.

File: .github/workflows/daily-testgrid.yaml (hunk 3 above)
Comment (paha) [reply .github/workflows/daily-testgrid.yaml:43]: Sounds good, the time/date reference is carried by other metadata. Changing in the next commit.

File: .github/workflows/deploy-testgrid.yaml [hunk 5]
```diff
@@ -86,7 +87,26 @@ jobs:
...
+          docker build -f testgrid/tgrun/Dockerfile \
+            -t replicated/tgrun:latest \
+            -t replicated/tgrun:${GITHUB_SHA:0:7} replicated/tgrun
+
+      - name: tgrun-push
+        env:
+          DOCKERHUB_DOCKER_PASS: ${{ secrets.DOCKERHUB_DOCKER_PASS }}
+          DOCKERHUB_DOCKER_USER: ${{ secrets.DOCKERHUB_DOCKER_USER }}
+        run: |
+           echo "$DOCKERHUB_DOCKER_PASS" | docker login -u $DOCKERHUB_DOCKER_USER --password-stdin
```
Comment (marccampbell) [inline .github/workflows/deploy-testgrid.yaml:108]: https://github.com/replicatedhq/troubleshoot/blob/master/.github/workflows/build-test-deploy.yaml#L105

Better to use the docker login action for this please

File: .github/workflows/deploy-testgrid.yaml [hunk 6]
```diff
@@ -86,7 +87,26 @@ jobs:
...
+           docker push replicated/tgrun
```
Comment (marccampbell) [inline .github/workflows/deploy-testgrid.yaml:109]: no tag?

**Review by marccampbell — CHANGES_REQUESTED** [review]

File: testgrid/tgrun/deploy/tgrun-pod.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: tgrun
+  labels:
+    app: testgrid
+spec:
+  containers:
+  - name: tgrun
+    image: replicated/tgrun
```
Comment (marccampbell) [inline testgrid/tgrun/deploy/tgrun-pod.yaml:10]: there should be a tag on this, it's going to be very hard to reproduce and troubleshoot without a tag

File: testgrid/tgrun/deploy/tgrun-pod.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: Pod
```
Comment (marccampbell) [inline testgrid/tgrun/deploy/tgrun-pod.yaml:2]: why not a deployment instead of a pod?

File: .github/workflows/daily-testgrid.yaml [hunk 9]
```diff
@@ -24,7 +24,7 @@ jobs:
 
     - name: testgrid-queue
       env:
-        TESTGRID_API: "https://api.testgrid.staging.kurl.sh/"
```
Comment (marccampbell) [inline .github/workflows/daily-testgrid.yaml]: we should run these 7 days a week ^^ the cron spec is 1-5 but images might change over the weekend

**Review by laverya — APPROVED** [review]
